mnefun/_mnefun.py

```diff
@@ -1417,18 +1418,17 @@ def save_epochs(p, subjects, in_names, in_numbers, analyses, out_names,
             # now make evoked for each out type
             evokeds = list()
             for name in names:
-                if p.weighting:
-                    #TODO give access to _chpi_snr_epochs args?
+                if p.average_weighting == 'chpi':
                     weights = _chpi_snr_epochs(e[name])
                     weights /= weights.sum()  # normalize weights
-                    weigh_these = [trl for trl in e[name].iter_evoked()]
+                    weigh_these = list(e[name].iter_evoked())
                     evoked = combine_evoked(weigh_these, weights=weights)
                     evoked.comment = name
```

Comment:
@Eric89GXL can you think of any reason why I would be losing epochs here? I am wondering whether it would not be safer to search and load `epo.fif` file rather than re-do epoching in order to prevent accidental changes in epoching being reintroduced into weighted averaging. WDYT?